node/src/chain.rs

```diff
@@ -391,6 +390,16 @@ pub async fn create_ethereum_networks_for_chain(
     }))
 }
 
+/// Networks as chains will create the necessary chains fromt he adapter information.
+/// There are two major cases that are handled currently:
+/// Non substreams chains(Ethereum, Near, etc etc):
+///  - These can have adapter of any type. Adapters of firehose and rpc types are used by the Chain implementation, aka deep integration
+///  - The substreams adapters will trigger the creation of a Substreams chain, this should only be used for block ingestors as a last resort.
+/// Substreams Chain:
+///  - This chain type is more generic and can only have adapters of substreams type.
+///  - Substreams chain are created as a "secondary" chain for deep integrations but in that case the block ingestor
+///    should be run by the main/deep integration chain.
```

Comment:
this is just a bit weird, since substreams is defined as both a substreams chain and an adapter type, if you have `protocol = Ethereum` on the toml then you need an rpc or firehose. If you have 'protocol = substreams' then you just need a substreams endpoint and cannot use any other kind.

Reply:
tried to improve the comment a bit, since this is somewhat urgent I will merge once it's green but if this is still unclear we can discuss how to better expose this information